core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -1164,11 +1167,7 @@ private Reader(Reader r, boolean useSample) {
 
     private void closeLocalityGroupReaders() {
       for (LocalityGroupReader lgr : currentReaders) {
-        try {
-          lgr.close();
-        } catch (IOException e) {
-          log.warn("Errored out attempting to close LocalityGroupReader.", e);
-        }
+        lgr.close();
```

Comment:
If `currentReaders` were changed to an ArrayList, this could use the stream `forEach`. Not sure if that would impact performance, but it'd probably improve readability in a few places.